docs/tutorials/gluon/performance.md

```diff
@@ -0,0 +1,483 @@
+
+## Tips & Tricks #4: Using Mixed-Precision (`float32` and `float16`)
+
+Model execution is still our bottleneck, so let's try out a new trick called [mixed precision](https://mxnet.incubator.apache.org/versions/master/faq/float16.html) training. Some recent GPUs have cores that are optimized for 'half-precision' (i.e. `float16`) operations and they can be much faster than their 'full-precision' (i.e. `float32`) counterparts. Given all of the randomness already in neural network training, this reduction in precision doesn't significantly impact the model accuracy in many cases. Convergence is slightly better when you keep the network parameters at full-precision but forward and backward passes can be performed at half-precision: hence the term 'mixed-precision'. Also check out [Automatic Mixed Precision](https://mxnet.incubator.apache.org/versions/master/tutorials/amp/amp_tutorial.html) (AMP) for a more automated way of optimizing your network.
+
+We need to `cast` the network to `'float16'`, configure our optimizer to use `multi_precision` and convert our input data types to `'float16'` too.
+
+
+```python
+net.cast('float16')
+trainer = mx.gluon.Trainer(net.collect_params(), 'sgd', {'learning_rate': learning_rate,
+                                                         'multi_precision': True})
+single_forward(net, dataloader, dtype='float16')
+```
+
+
+```python
+print('\n', '### benchmark_dataloader', '\n')
+benchmark_dataloader(dataloader)
+print('\n', '### benchmark_network', '\n')
+data, label = next(iter(dataloader))
+data = data.astype('float16').as_in_context(ctx)
+label = label.astype('float16').as_in_context(ctx)
+benchmark_network(data, label, net, loss_fn, trainer)
+```
+
+    
+     ### benchmark_dataloader 
+    
+    ........................
+    
+    total startup time: 0.7095
+    average iterations/sec: 11.5624
+    average samples/sec: 739.9948
+    
+     ### benchmark_network 
+    
+    ........................
+    
+    average iterations/sec: 8.5895
+    average samples/sec: 549.7281
+
+
+Overall we see a substantial increase in training throughput: ~85% higher than full-precision training.
+
+## Tips & Tricks #5: Others
+
+Many other tips and tricks exist for optimizing the throughput of training.
+
+One area we didn't explicitly benchmark in this tutorial is data transfer from CPU to GPU memory. Usually this isn't an issue, but for very large arrays this can become a bottleneck too. You might be able to compress your data significantly before transferring if your data is sparse (i.e. mostly zero values). Check out the sparse array tutorial for more details and an example of how this can impact training speed.
+
+Another useful trick if data pre-processing or data transfer is the bottleneck is pre-fetching batches. You can write your training loop to transfer the next batch of data to GPU before processing the current batch. Once again, this trick is memory permitting.
```

Comment:
This would actually be very nice to show in this tutorial, since when using the CPU loader it is pretty much free gain (the only resource used is the CPU memory).

Reply:
Quite a complex trick for a beginner tutorial, will leave out details for now. Can follow up with a more advanced performance post later on, since there are many other tricks not discussed here.